server/graphql/stocktake_line/src/stocktake_line_queries.rs

```diff
@@ -0,0 +1,337 @@
+use repository::*;
+use serde::Serialize;
+use service::auth::Resource;
+use service::auth::ResourceAccessRequest;
+use service::stocktake_line::query::GetStocktakeLinesError;
+use service::ListError;
+use strum::{EnumIter, IntoEnumIterator};
+
+#[derive(InputObject, Clone)]
+pub struct StocktakeLineFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub stocktake_id: Option<EqualFilterStringInput>,
+    pub location_id: Option<EqualFilterStringInput>,
+}
+
+impl From<StocktakeLineFilterInput> for StocktakeLineFilter {
+    fn from(f: StocktakeLineFilterInput) -> Self {
+        StocktakeLineFilter {
+            id: f.id.map(EqualFilter::from),
+            stocktake_id: f.stocktake_id.map(EqualFilter::from),
+            location_id: f.location_id.map(EqualFilter::from),
+        }
+    }
+}
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq, Serialize, EnumIter)]
+#[graphql(rename_items = "camelCase")]
+pub enum StocktakeLineSortFieldInput {
+    ItemCode,
+    ItemName,
+    /// Stocktake line batch
+    Batch,
+    /// Stocktake line expiry date
+    ExpiryDate,
+    /// Stocktake line pack size
+    PackSize,
+    /// Stocktake line item stock location name
+    LocationName,
+}
+
+#[derive(InputObject)]
+pub struct StocktakeLineSortInput {
+    /// Sort query result by `key`
+    pub key: StocktakeLineSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    pub desc: Option<bool>,
+}
+
+impl StocktakeLineSortInput {
+    pub fn to_domain(self) -> StocktakeLineSort {
+        use StocktakeLineSortField as to;
+        use StocktakeLineSortFieldInput as from;
+        let key = match self.key {
+            from::ItemCode => to::ItemCode,
+            from::ItemName => to::ItemName,
+            from::Batch => to::Batch,
+            from::ExpiryDate => to::ExpiryDate,
+            from::PackSize => to::PackSize,
+            from::LocationName => to::LocationName,
+        };
+
+        StocktakeLineSort {
+            key,
+            desc: self.desc,
+        }
+    }
+}
+
+#[derive(Union)]
+pub enum StocktakesLinesResponse {
+    Response(StocktakeLineConnector),
+}
+
+pub fn stocktake_lines(
+    ctx: &Context<'_>,
+    store_id: &str,
+    page: Option<PaginationInput>,
+    filter: Option<StocktakeLineFilterInput>,
+    sort: Option<StocktakeLineSortInput>,
+) -> Result<StocktakesLinesResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::QueryStocktake,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_ctx = service_provider.context(store_id.to_string(), user.user_id)?;
+    let service = &service_provider.stocktake_line_service;
+
+    let stocktake_filter = filter.clone().map(StocktakeLineFilter::from);
+
+    let mut stocktake_id = "".to_string();
```

Comment:
I think instead of filter this should be graphql query parameter. This should simplify this a little bit

Reply:
Yeah could simplify that but I thought to handle it through the filter so that this graphql endpoint becomes reusable for any cases? Making `stocktake_id` parameter to this method makes this endpoint stocktake id specific (where we might not be able to just use this endpoint to get just stocktake lines - not sure where we use it thought).

Reply:
I had similar thought, but I couldn't quite figure out the use case for it, plus the service requires stocktake_id. I think if we need it in the future we can make another end point for all stocktake lines. 
Do you mind changing to stocktake_id in query parameters for now, and we can explore other options in the future, when the use case arrises, thanks !

Reply:
f9f96d248